opentracing/tracer.py

```diff
@@ -40,31 +41,84 @@ class Tracer(object):
     def __init__(self):
         self._noop_span_context = SpanContext()
         self._noop_span = Span(tracer=self, context=self._noop_span_context)
+        self._active_span_source = NoopActiveSpanSource()
 
-    def start_span(self,
-                   operation_name=None,
-                   child_of=None,
-                   references=None,
-                   tags=None,
-                   start_time=None):
+    @property
+    def active_span_source(self):
+        return self._active_span_source
+
+    @property
+    def active_span(self):
+        return self._active_span_source.active_span
+
+    def start_active_span(self, operation_name, tags=None, start_time=None):
+        """Starts and returns a new `Span` representing a unit of work. This
+        method differs from `start_manual_span` because it uses in-process
+        context propagation to keep track of the current active `Span` (if
+        available).
+
+        Starting a root `Span` with no casual references and a child `Span`
+        in a different function, is possible without passing the parent
+        reference around::
+
+            def handle_request(request):
+                root_span = tracer.start_active_span('request.handler')
+                data = get_data(request)
+
+            def get_data(request):
+                # child_span has `root_span` as a parent
+                child_span = tracer.start_active_span('db.query')
+
+
+        :param operation_name: name of the operation represented by the new
+            span from the perspective of the current service.
+        :param tags: an optional dictionary of Span Tags. The caller gives up
+            ownership of that dictionary, because the Tracer may use it as-is
+            to avoid extra data copying.
+        :param start_time: an explicit Span start time as a unix timestamp per
+            time.time()
+
+        :return: Returns an already-started `Span` instance, marked as active.
+        """
+        # use an ActiveSpanSource to retrieve the current Span
+        parent_span = self.active_span
```

Comment:
In general, I agree with you. I can make a pass to remove those properties! Thanks for the review!